docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,715 @@
+   cleaner to store it in `Status` but technically that can be lost, whereas `Annotations` won't be. This should work for both KeyVault and k8s secrets and 
+   while it's a bit icky I think gives the best experience.
+
+Conclusion: TBD
+
+**Should we drop the `Destination` suffix in the field names?**
+
+For example, the proposal for CosmosDB `DatabaseAccount` was:
+```yaml
+spec:
+  # Other spec fields elided...
+  forOperator:
+    secrets:
+      keyDestination:
+        # stuff elided...
+      readOnlyKeyDestination:
+        # stuff elided...
+```
+
+```yaml
+spec:
+  # Other spec fields elided...
+  forOperator:
+    secrets:
+      key:
+        # stuff elided...
+      readOnlyKey:
+        # stuff elided...
+```
+
+Conclusion: TBD
+
+**Do we allow writing a secret to a namespace where the resource isn't?**
+
+No, see: https://github.com/kubernetes/community/pull/5455
+
+Conclusion: TBD
+
+**Do we support writing the same secret to multiple destinations?**
+
+If/when we add support for writing secrets to KeyVault rather than to Kubernetes secrets, would we also enable support for writing the same secret to _both_
+KeyVault and Kubernetes? This would allow the user to specify a `spec` like:
+```yaml
+      # Save the read-write keys (and endpoints) into a kubernetes secret called "my-secret" and a KeyVault secret called "my-secret".
+      keyDestination:
+        kubernetes:
+          name: my-secret
+        keyVault:
+          reference:
+            armId: /subscriptions/.../resourceGroups/.../providers/Microsoft.KeyVault/vaults/asokeyvault
+          name: my-secret
+```
+
+I think we should.
+
+Conclusion: TBD
+
+**What client should we use for issuing the ListKeys/GetKeys request?**
+
+We don't automatically generate methods for performing these calls. It seems easiest to just use the corresponding Azure SDK for this.
+This does mean that we're taking a dependency on the SDK where we didn't have one before, but the overall footprint of usage is pretty small
+and it's going to be easier to do that than it is to write the methods ourselves.
+
+Conclusion: TBD
+
+**What API version should we use for issuing the ListKeys/GetKeys request?**
+
+There are a few options here. The custom hooks could be applied to either the customer facing resource version _or_ to the internal storage version.
+Applying the hooks to the storage version is easier, but means that the same hook will be run regardless of which customer 
+
+#### Alternative design
+
+Would we prefer an alternative design where each individual field that goes into a secret is called out explicitly? It's noisier but more flexible and also 
+probably more discoverable, at least in theory? Here's what it might look like for CosmosDB `DatabaseAccount`
+
+```yaml
+  # Other spec fields elided...
+  forOperator:
+    secrets:
+      primaryKeyDestination:
```

Comment:
Maybe this should be a list with names rather than a map, for better compliance with the k8s guidelines? I guess we still have a lot of maps that we inherit from ARM, but it's probably worth not making it worse.

Reply:
My intent at least was that it is an object, not an open ended map.